docs-src/concepts/what-is-versioning.md

```diff
@@ -0,0 +1,170 @@
+
+### Disadvantages of Patching
+
+Patching has several drawbacks:
+
+- **Conceptually complex**: While the code to write a patch is simple, the conceptual nature of patching is a cognitive burden of needing to understand how both the _old_ and _new_ code paths work.
+- **Increased maintenance**: As your Workflow code paths start to branch, tracking the state of each branch becomes more complex. This is especially true if you need to patch a Workflow that has already branched. For example, Entity Workflows or indefinitely running Workflows complicate the patching process when you start to patch a Workflow that has already branched.
+
+## Workflow Type Versioning
+
+Workflow Type versioning is a technique that involves creating a new Workflow Type for every iteration or version of your Workflow. This strategy represents the most basic form of Workflow Versioning.
+
+### What is Workflow Type Versioning?
+
+In essence, Workflow Type versioning is the process of delineating different versions of your Workflow by associating them with distinct Workflow Types. For instance, when you change a Workflow code path, you update the Workflow Type as the next iteration in a sequence of Workflow Types: `workflow-v1`, `workflow-v2`, `workflow-v3`.
+
+This approach offers a clear and straightforward way of distinguishing between various versions of your Workflow.
+
+### Implement Workflow Type Versioning
+
+To implement Workflow Type versioning, you need to create a new Workflow Type each time you intend to deploy an updated version of your Workflow. For instance, upon deploying `workflow-v1`, if you need to make changes, you would create a new Workflow Type and designate it as `workflow-v2`. The process continues in the same vein for all subsequent versions, such as `workflow-v3`, `workflow-v4`, and so forth.
+While this approach is simple, it does require you to manually update all starters to use the latest Workflow Type. Additionally, you need to keep track of all Workflow Types and their corresponding versions.
+
+### Advantages of Workflow Type Versioning
+
+Workflow Type versioning is valuable in several respects:
+
+- **Simplicity:** Workflow Type versioning provides a straightforward way to version your Workflows, making it an excellent choice for uncomplicated or small-scale projects.
+- **Single Worker Fleet:** With this approach, you're not required to manage separate Worker fleets for each version of your Workflow, potentially minimizing infrastructure complexity and cost.
+
+### Disadvantages of Workflow Type Versioning
+
+Workflow Type versioning has several drawbacks, including:
+
+- **Version Tracking:** All starters need to be informed about each new Workflow Type and its corresponding version, which might introduce communication overhead and room for error.
+
```

Comment:
Lots of code duplication is probably the biggest disadvantage and we should mention that.